exercises/palindrome-products/test/test_palindrome_products.c

```diff
@@ -9,24 +12,45 @@ void tearDown(void)
 {
 }
 
-void check_factors(factor_t * actual, int depth, factor_t expected[])
+bool factor_t_are_equal(const factor_t * const f1, const factor_t * const f2)
+{
+   return ((f1->factor_a == f2->factor_a) && (f1->factor_b == f2->factor_b)) ||
+       ((f1->factor_a == f2->factor_b) && (f1->factor_a == f2->factor_b)
+       );
+}
+
+bool contains_factor(const factor_t factors[], const size_t depth,
+                     const factor_t * const factor)
+{
+   for (size_t i = 0; i < depth; ++i) {
+      if (factor_t_are_equal(&factors[i], factor)) {
+         return true;
+      }
+   }
+   return false;
+}
+
+void check_factors(factor_t * actual, size_t depth, factor_t expected[])
```

Comment:
Does this work if we pass in this?

expected == {{1, 9}, {3, 3}}
actual == {{1, 9}, {1, 9}}

Reply:
This test case passes with the current version on the `check_factors`. Will try to fix this.

Reply:
@patricksjackson I think I managed to fix the `check_factors` behavior, can you give it another look?

Reply:
Yep, should work now with any order. It won't give very good error messages, but I'm not quite sure how to improve that.

(I admittedly am a bit rusty on my C, so I didn't expect it to be this complicated. In C++ it'd be a lot easier to just throw the LL into an array, sort everything, and compare them. That approach might be a bit difficult for C though.)